jck/jck-semiauto-updater/jckupdater.sh

```diff
@@ -167,50 +171,126 @@ isLatestUpdate() {
+		fi
 		getJCKSources
 	fi
 }
 
+getLastModifiedForFile() {
+    file="$1"
+    curl -s -H "X-JFrog-Art-Api:$ARTIFACTORY_TOKEN" "$ARTIFACTORY_API_URL$file" | awk -F'"' '/lastModified/ {print $4}'
+}
+
+getExcludeFiles(){
+	github_last_modified=$(curl -s -H "Authorization: token $GIT_TOKEN" "https://api.github.ibm.com/repos/runtimes/JCK$JCK_VERSION-unzipped/commits?path=excludes/jck$JCK_FOLDER_SUFFIX.jtx" | awk -F'"date":' '{if(NF>1){gsub(/[",]/,"",$2); print $2; exit}}')
+	#check if file is present in github.
+	if [[ -n "${github_last_modified}" ]]; then
+		github_date_only=$(date -u -d "${github_last_modified}" "+%Y-%m-%d")
+	fi
+	artifactory_file_list=$(curl -s -H "X-JFrog-Art-Api:$ARTIFACTORY_TOKEN" "$ARTIFACTORY_API_URL" -s | \
+    awk -F'"uri" : "' '/"uri" : "\// { gsub(/".*$/, "", $2); print $2 }' | \
+    awk '{gsub(/^\//, ""); print}' | \
+    grep -v '/$' )
+
+	#if no exclude file is present
+	if [[ -z "$artifactory_file_list" ]]; then
+		echo "No exclude files present in Artifactory to update"
+		return 0
+	else
+		declare -a last_modified_dates
+		declare -a file_last_modified
+
+		echo "List of exclude files in Artifactory:"
+		echo "$artifactory_file_list"
+
+		#store all files one by one
+		files=()
+		while IFS= read -r line; do
+			files+=("$line")
+		done <<< "$artifactory_file_list"
+
+		for file in "${files[@]}"; do
+			artifactory_date=$(date -d "$(getLastModifiedForFile "$file")" "+%Y-%-m-%d")
+			last_modified_dates+=("$artifactory_date")
+			file_last_modified["$artifactory_date"]=$file
+		done
+
+		artifactory_latest_modified=$(printf "%s\n" "${last_modified_dates[@]}" | sort -rV | head -1)
+		latest_file_name="${file_last_modified[$artifactory_latest_modified]}"
+		formatted_date=$(date -d "$artifactory_latest_modified" "+%Y-%m-%d") # reconvert month to value with leading zero for comparison
+		echo "Latest exclude file: $latest_file_name"
+		echo "Last Modified Date in GitHub: ${github_date_only}"
+		echo "Last Modified Date in Artifactory: ${formatted_date}"
+
+		# Compare and find the latest last modified date only if file is present in github. Else update exclude file
+		if [[ -n "${github_last_modified}" && "${github_date_only}" > "${formatted_date}" ]]; then
+			echo "No need to update exclude file in GitHub"
+			return 0
+		else
+			echo "Update exclude file with $latest_file_name from Artifactory"
+			return 1
+		fi
+	fi
+
+}
+
 ## Download directly from given URL under current folder
 getJCKSources() {
 	cd $WORKSPACE/jckmaterial
 	echo "remove build.txt file after comparison"
 	rm -rf build.txt
 	echo "download jck materials..."
 	
-	ARTIFACTORY_DOWNLOAD_URL=$ARTIFACTORY_DOWNLOAD_URL$JCK_UPDATE_NUMBER
-	echo $ARTIFACTORY_DOWNLOAD_URL
+	#download latest exclude file
+	if [ $exclude_files -eq 1 ]; then
+		executeCmdWithRetry "${latest_file_name##*/}" "_ENCODE_FILE_NEW=UNTAGGED curl -OLJSk -H X-JFrog-Art-Api:${ARTIFACTORY_TOKEN} ${ARTIFACTORY_DOWNLOAD_URL}exc/$latest_file_name"
```

Comment:
I do not think `executeCmdWithRetry` is needed here. Please remove it. If you want to use it, then we need to make it util function. Not copy and paste the code.